### PR TITLE
Provide a tool for generation of integration tests

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Provides a developer tool to generate a skeleton for integration tests.

## Motivation and additional information

We want to restructure the integration tests and this tool will help us. See #1021.

This PR was extracted from #1140, for more details, please refer to #1140.

## Author's checklist

* [ ] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
* [ ] I ran `tools/formatting/format-all` to ensure everything is formatted correctly.
* [ ] I sticked to C++14 features.
* [ ] I sticked to CMake version 3.10.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: tools/formatting/format-all [hunk 1]
```diff
@@ -6,6 +6,8 @@
 # 0 on success
 # 2 is clang-format 8 could not be found 
 
+#TODO: Needs an update?
+
```
Comment (BenjaminRodenberg) [inline tools/formatting/format-all:10]: @fsimonis Could you check this during the review?

**Comment by BenjaminRodenberg** [comment]
When cleaning up #1140, I realized that there is still a bug in this script:
```
precice/tests$ python3 ../tools/building/createTest.py Serial/MultiCouplingFourSolvers1
usage: createTest.py [-h] [-n] [Suite/]TestName
createTest.py: error: argument [Suite/]TestName: The given test name "MultiCouplingFourSolvers1" must use CamelCase.
```
It seems like `MultiCouplingFourSolvers1` is not properly detected as CamelCase.

File: tools/formatting/format-all (hunk 1 above)
Comment (fsimonis) [reply tools/formatting/format-all]: Removed it. The formatting scripts don't need an update.

**Review by fsimonis — APPROVED** [review]

**Comment by fsimonis** [comment]
Let me clean this up. The commit history is too wild for my liking.

**Comment by uekerman** [comment]
@BenjaminRodenberg Did you already document how to use this tool? Should be part of dev docs, I guess.

**Pull request merged**